tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/TspTool/TspTool.cs

```diff
@@ -0,0 +1,342 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.ComponentModel;
+using Azure.Sdk.Tools.Cli.Services;
+using Azure.Sdk.Tools.Cli.Contract;
+using Azure.Sdk.Tools.Cli.Models.Responses;
+using ModelContextProtocol.Server;
+using Azure.Sdk.Tools.Cli.Helpers;
+
+namespace Azure.Sdk.Tools.Cli.Tools.TspTool
+{
+    /// <summary>
+    /// This tool provides functionality for initializing TypeSpec projects and converting existing Azure service swagger definitions to TypeSpec projects.
+    /// Use this tool to onboard to TypeSpec for new services or convert existing services.
+    /// </summary>
+    [McpServerToolType, Description("Tools for initializing TypeSpec projects and converting existing Azure service swagger definitions to TypeSpec projects.")]
+    public class TypeSpecTool(IProcessHelper processHelper, ILogger<TypeSpecTool> logger, IOutputService output) : MCPTool
+    {
+
+        // commands
+        private const string initCommandName = "init";
+        private const string convertSwaggerCommandName = "convert-swagger";
+
+        private readonly Argument<string> templateArg = new("--template", "The template to use for the TypeSpec project. Valid values are: azure-core (for data-plane services), azure-arm (for resource-manager services)");
+        private readonly Argument<string> serviceNamespaceArg = new("--service-namespace", "The namespace of the service you are creating. This should be in Pascal case and represent the service's namespace.");
+        private readonly Argument<string> outputDirectoryArg = new("--output-directory", "The output directory for the generated TypeSpec project. This directory must already exist and be empty.");
+
+        private readonly Argument<string> swaggerReadmeArg = new("--swagger-readme", "The path or URL to an Azure swagger README file.");
+        private readonly Option<bool> isArmOption = new("--arm", "Whether the generated TypeSpec project is for an Azure Resource Management (ARM) API. This should be true if the swagger's path contains 'resource-manager'.");
+        private readonly Option<bool> fullyCompatibleOption = new("--fully-compatible", "Whether to generate a TypeSpec project that is fully compatible with the swagger. It is recommended not to set this to true so that the converted TypeSpec project leverages TypeSpec built-in libraries with standard patterns and templates.");
+
+        public override Command GetCommand()
+        {
+            var tspCommand = new Command("tsp", "Tools for initializing TypeSpec projects and converting existing Azure service swagger definitions to TypeSpec projects");
```

Comment:
We have some abstraction to enable sub-commands (https://github.com/Azure/azure-sdk-tools/blob/a2fe9df6c47d068dac9cc2cef88f97f7cbc645af/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/EngSys/Cleanup/CleanupTool.cs#L30-L43, https://github.com/Azure/azure-sdk-tools/blob/a2fe9df6c47d068dac9cc2cef88f97f7cbc645af/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Commands/SharedCommandGroups.cs#L8) that you can use. I need to come up with an example for you though.

Reply:
Ah - I think I get it - I can create a `tsp` shared command, and that's the entry I'd have in my `CommandHierarchy`. So then I wouldn't have to do the `tspCommand.AddCommand` call.

Is that the main advantage? Would we want all commands that have more than 1 subcommand to be in `SharedCommandGroups`, even if all subcommands are in one file?

Reply:
The general idea with the command hierarchy is that we can know with a glance what exact command a tool class corresponds to. Additionally when we do have sub-commands across classes we can share options and arguments vs. having to re-define them in each class.

I don't think we should mandate that each sub-command go in its own class as that seems tedious. But trying to figure out whether there's a middle ground here. @scbedd any thoughts?

Reply:
The goal of the `command hierarchy` is so that you can get grouped subcommands across various command classes without having to do a bunch of work to ensure that those two subcommands belong to the same parent, which then gets picked up by the automation creating the CLI entrypoint for the server.

IF you're just adding the subcommands to the parent command that gets returned by `GetCommand()`, I don't see where this would break the [standard command hierarchy creation.](https://github.com/Azure/azure-sdk-tools/blob/c2862713609a6dbe51aa65f555ffe07832bb3707/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Commands/CommandFactory.cs#L46).